enricher/fabric8/src/main/java/io/fabric8/maven/enricher/fabric8/DockerHealthCheckEnricher.java

```diff
@@ -0,0 +1,124 @@
+package io.fabric8.maven.enricher.fabric8;
+
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+
+import io.fabric8.kubernetes.api.model.Container;
+import io.fabric8.kubernetes.api.model.ExecAction;
+import io.fabric8.kubernetes.api.model.Probe;
+import io.fabric8.kubernetes.api.model.ProbeBuilder;
+import io.fabric8.kubernetes.api.model.extensions.Deployment;
+import io.fabric8.maven.core.util.KubernetesResourceUtil;
+import io.fabric8.maven.docker.config.HealthCheckConfiguration;
+import io.fabric8.maven.docker.config.HealthCheckMode;
+import io.fabric8.maven.docker.config.ImageConfiguration;
+import io.fabric8.maven.enricher.api.AbstractHealthCheckEnricher;
+import io.fabric8.maven.enricher.api.EnricherContext;
+
+import static io.fabric8.maven.enricher.api.util.GoTimeUtil.durationSeconds;
+
+/**
+ * Enrich a container with probes when health checks are defined in the {@code ImageConfiguration} of the docker maven plugin.
+ * This enricher could need a change when Dockerfile health checks will be supported natively in Openshift or Kubernetes.
+ */
+public class DockerHealthCheckEnricher extends AbstractHealthCheckEnricher {
+
+    public DockerHealthCheckEnricher(EnricherContext buildContext) {
+        super(buildContext, "docker-health-check");
+    }
+
+    /**
+     * Try to provide a health check for any image defined in the config.
+     */
+    @Override
+    protected List<Container> getCandidateContainers(Deployment deployment) {
```

Comment:
Not really related to this PR, but I wonder whether we should support `DeploymentConfig` here, too ? (and also `ReplicaSet`, `ReplicationController` and finally `PetSet`, `DaemonSet` ....).

If this is a good idea, I would open a corresponding issue for it.